src/Microsoft.Extensions.Logging.Log4Net.AspNetCore/Log4NetLogger.cs

```diff
@@ -108,22 +101,16 @@ public bool IsEnabled(LogLevel logLevel)
             LogMessage(message);
         }
 
-        private void LogMessage(MessageCandidate candidate)
-        {
-            if (candidate.IsValid())
-            {
-                var level = SelectLevel(candidate);
-                if (level != null)
-                {
-                    Log(candidate, level);
-                }
-            }
-        }
-
-        private Level SelectLevel(MessageCandidate candidate)
+        /// <summary>
+        /// Translates a <see cref="LogLevel"/> to a log4net <see cref="Level"/> based on the provided options.
+        /// </summary>
+        /// <param name="logLevel">The log level to translate.</param>
+        /// <param name="options">The log4net provider options influencing the translation.</param>
+        /// <returns>The corresponding log level for log4net.</returns>
+        public static Level GetTranslatedLog4NetLevel(LogLevel logLevel, Log4NetProviderOptions options)
```

Comment:
Is there any reason to expose this method as public? :)

Reply:
If you wan't to build your own LoggingEventFactory you would need to do the LogLevel translation yourself and can't use the default.

Reply:
Then, taking advantage of this change, could be a good idea to move the feature into separated type, in order to keep the SRP principle for the consumer, and inject it somehow to this logger class when created?

Reply:
ok I did something like that. let's see if you like it 😄